src/txmempool.cpp

```diff
@@ -387,6 +387,14 @@ void CTxMemPool::removeWithAnchor(const uint256 &invalidRoot, ShieldedType type)
                     }
                 }
             break;
+            case ORCHARD:
+            {
+                auto anchor = tx.GetOrchardBundle().GetAnchor();
+                if (anchor == invalidRoot) {
```

Comment:
It does indeed seem possible to compare a value with type `std::optional<T>` to a value with type `T`; I presume this is because `invalidRoot` is being implicitly converted to `std::optional(invalidRoot)` in the argument to `operator==(std::optional<T> bar)`, like happens when we pass these values into other methods that accept `std::optional<T>`.